queries.sql

```diff
@@ -1,10 +1,68 @@
 
-SELECT * FROM vet_clinic.animals WHERE name like '%mon'
+SELECT * FROM vet_clinic.animals WHERE name LIKE '%mon'
 SELECT name FROM vet_clinic.animals WHERE date_of_birth >= '20160101' AND date_of_birth <= '20191231';
 SELECT name FROM vet_clinic.animals WHERE neutered = true AND escape_attempts < 3;
 SELECT date_of_birth FROM vet_clinic.animals WHERE name = 'Agumon' OR name = 'Pikachu';
 SELECT name, escape_attempts FROM vet_clinic.animals WHERE weight_kg > "10.50";
 SELECT * FROM vet_clinic.animals WHERE neutered = true;
 SELECT * FROM vet_clinic.animals WHERE name != "Gabumon";
-SELECT * FROM vet_clinic.animals WHERE weight_kg >= "10.40" AND weight_kg <= "17.30";
\ No newline at end of file
+SELECT * FROM vet_clinic.animals WHERE weight_kg >= "10.40" AND weight_kg <= "17.30";
+
+/*Inside a transaction update the animals table by setting the species column to unspecified*/
+BEGIN;
+UPDATE vet_clinic.animals SET species = 'unspecified' WHERE id > 0;
+ROLLBACK;
+
+/*Inside a transaction:
+Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+Commit the transaction.
+Verify that change was made and persists after commit.*/
+
+BEGIN;
+UPDATE vet_clinic.animals SET species = 'digimon' WHERE name LIKE '%mon' AND id > '0';
+UPDATE vet_clinic.animals SET species = 'pokemon' WHERE name NOT LIKE '%mon' AND id > '0';
+COMMIT;
+
+/*Inside a transaction delete all records in the animals table, then roll back the transaction.*/
+
+BEGIN;
+DELETE FROM vet_clinic.animals WHERE id > 0;
+ROLLBACK;
+
+/*Inside a transaction:
+Delete all animals born after Jan 1st, 2022.
+Create a savepoint for the transaction.
+Update all animals' weight to be their weight multiplied by -1.
+Rollback to the savepoint
+Update all animals' weights that are negative to be their weight multiplied by -1.
+Commit transaction*/
+
+BEGIN;
+DELETE FROM vet_clinic.animals WHERE date_of_birth > 20220101 AND id > '0';
+SAVEPOINT delete_from_birth;
+UPDATE vet_clinic.animals SET weight_kg = weight_kg*-1 WHERE id > '0';
+ROLLBACK TO SAVEPOINT delete_from_birth;
+UPDATE vet_clinic.animals SET weight_kg = weight_kg*-1 WHERE weight_kg < 0 AND id > '0';
+COMMIT;
+
+
+/*Write queries to answer the following questions:
+How many animals are there?
+How many animals have never tried to escape?
+What is the average weight of animals?
+Who escapes the most, neutered or not neutered animals?
+What is the minimum and maximum weight of each type of animal?
+What is the average number of escape attempts per animal type of those born between 1990 and 2000?*/
+
+SELECT COUNT(*) FROM vet_clinic.animals;
+SELECT COUNT(*) FROM vet_clinic.animals WHERE escape_attempts > 0;
+SELECT AVG(weight_kg) FROM vet_clinic.animals;
+SELECT MAX(escape_attempts) FROM vet_clinic.animals;
```

Comment:
* Kindly add the right query as the question says ( Who escapes the most, neutered or not neutered animals?) i think your query can revolve around ( neutered and non neutered animal ) this can be achieved by selecting and grouping your query by neutered from the `Animal ` table.